server/service/oauth.go

```diff
@@ -68,16 +68,18 @@ type GithubUser struct {
 }
 
 // MakeAuthURL returns string for redirect to provider
-func (o *OAuth) MakeAuthURL(w http.ResponseWriter, r *http.Request) string {
+func (o *OAuth) MakeAuthURL(w http.ResponseWriter, r *http.Request) (string, error) {
 	b := make([]byte, 16)
 	rand.Read(b)
 	state := base64.URLEncoding.EncodeToString(b)
 
 	session, _ := o.store.Get(r, "sess")
```

Comment:
if you return `(string, error)`, it makes sense to handle error here too.

Reply:
yesssss, you're totally right; thanks for spotting it!